src/lozad.js

```diff
@@ -35,22 +35,17 @@ export default function (selector = '.lozad', options = {}) {
 
   return {
     observe() {
-      const elements = [].filter.call(document.querySelectorAll(selector),
-      element => !isLoaded(element))
-
-      if (!observer) {
-        elements
-          .forEach(element => {
-            load(element)
-            markAsLoaded(element)
-          })
-
-        return
+      const elements = document.querySelectorAll(selector)
+      for (let i = 0; i < elements.length; i++) {
```

Comment:
What do you think about using early returns to avoid the else, improve readability and decrease the indentation levels?
Something like:

```js
for (let i = 0; i < elements.length; i++) {
  if (isLoaded(elements[i])) {
    return
  }

  if (observer) {
    observer.observe(elements[i])
    return
  }

  load(elements[i])
  markAsLoaded(elements[i])
}
```

Reply:
Yep that works too. New commit coming

Reply:
Awesome! I forgot that the `return` will not work. Lol.